### DISCUSSION TITLE
Resolution

### DISCUSSION THREAD

**Discussion by highaltidude** [blakeblackshear/frigate#1189] [Ask A Question] [answered]
So if models are in a 300 x 300px images, why does having a higher resolution offer no performance increase? Let's say I have a 4k camera, there should be a huge canvas for analytics to find objects. If Frigate reduces the area to 300x300 doesn't that make the detect range of Frigate extremely limited? Love the project. Thanks!

**Comment by blakeblackshear** [comment, marked as answer]
A higher resolution _model_ would in theory offer better performance, but likely wont fit on the Coral or be performant enough for realtime analysis. These aren't frigate-specific models. They are provided by Google, and they are 300x300 or 320x320. 

> Let's say I have a 4k camera, there should be a huge canvas for analytics to find objects

This is why frigate doesn't pass the entire frame to object detection. It looks for motion and then creates a bounding box around a subset of the frame and sends that for object detection. That bounding box area is always resized to the model size (320x320). When running detection on a 4k stream, the bounding box for motion when a person is near will likely be _much_ larger than 320x320. Resize operations are always a balance of speed and data loss, and frigate prioritizes speed, so the more you shrink the image, the worse the image quality. Of course if you are using a 4k stream to detect very small, far away objects, this won't be a problem.

**Reply by highaltidude** [reply]
Make sense, Thanks for the clarification! Should be getting my Google Coral in the next few days so I can add a few more cameras. Those suckers are hard to find!

**Reply by rhatguy** [reply]
Given that frigate is sending a subset of the frame (where it detected motion) and using that for analysis...if we are using a 640x480 stream for detection does that mean that unless the bounding box takes up ~half the width of the frame, then we're down sampling and loosing fidelity?  I.E. would it be better to shoot for cutting a 300x300 bounding box out of a 1920x1080 frame?  I realize its not a perfect science and we can't ensure how big the bounding box will be.  It seems like cutting a 300x300 motion box out of a 640x480 sub-stream the motion box would result in a lot of downsampling.

**Reply by blakeblackshear** [reply]
Frigate won't select a bounding box smaller than 320x320 (or some minimum). If a person takes up most of the frame of a 640x480, then this is about perfect because the size of the bounding box is already close to the model size. If it was a 1080p stream, the box may be 1080x1080 and require shrinking to 320x320. Given the same camera placement, I would expect the 1080p feed to perform worse for up close objects.

**Reply by poweredgenl** [reply]
so when i understand you correctly - het goal is to balance between res and distance of the person on screen... eg if people where to be further away you need the more pixels to get to a suitable 300x300 model? so if ppl would be further away on the camera ... a higher resolution is more beneficial?

**Reply by blakeblackshear** [reply]
Yes

**Reply by poweredgenl** [reply]
ok clear - then im getting the hang of it - in my case with wide view security cams - a higher reso would be better - eg limit fps to not overstretch the RPI... and then in MQTT as i use frigate with compreface as well increase the snapshot it will sent to MQTT ? would this be also your approach?

**Reply by blakeblackshear** [reply]
In general, I don't consider MQTT to be a great way to get an image for further processing. You can't reliability connect the snapshot to an event id. I would use the events mqtt topic and fetch images from the http api as needed for further processing.

**Reply by haggercody** [reply]
i got a coral so i could test out using my main stream for detect because our camera is pretty wide angle and resolution really drops when you get just a bit away from our house so we've never been able to motion detect there with blue iris without getting a ton of noise.  Frigate is doing an awesome job on this.   The only downside i've noticed is that it appears there is an ffmpeg process that is running a bit more cpu than before--my guess is the detect stream.  seems to hover about 20% on one cpu.
